### DISCUSSION TITLE
Analyzer Data Flashes Then Disappears

### DISCUSSION THREAD

**Discussion by msevette** [artisan-roaster-scope/artisan#1736] [General]
I'm running Mac 11.7.10 and Artisan v3.0.2. When I use the analyzer tools, the data boxes flash and then disappear. However, the curve lines remain. This did not happen in v2.8.4. I have no problem going back to v2.8.4, but did not keep the .dmg and cannot find where to download it. Please help.

**Comment by MAKOMO** [comment]
Could you please attach the artisan profile where this happens as .alog file (rename to end in .txt and drag-and-drop it to a dialog box here). What are the exact instructions to reproduce this? Load the profile and select Analyzer >> Fit BT to x^2 ?

We will investigate and fix if possible. Thanks for your report!

**Reply by msevette** [reply]
I have attached the profile as .txt. Load the profile and select Tools >> Analyzer >> Auto All. Actually, it does not matter which analyzer tool is selected; the data boxes disappear on all. Also, the projected curve line does not disappear when clear is selected.

Thank you for looking into this.

[v2-046_guatemala-xinabajul-los-chuchitos_24-10-27_1408.txt](https://github.com/user-attachments/files/17630731/v2-046_guatemala-xinabajul-los-chuchitos_24-10-27_1408.txt)

**Reply by MAKOMO** [reply]
Thanks for this file. I just loaded it in Artisan v3.0.2 and it just works fine for me. Could you please also upload your Artisan settings that we can reproduce your exact Artisan state to reproduce this. For this, save your Artisan settings via menu Help >> Save Settings into an artisan-settings.aset file, rename that one to end in .txt and again drag-and-drop it here on Github on a comment window.

**Reply by msevette** [reply]
Here is my Artisan settings file:

[artisan-settings-mac11-20241105.txt](https://github.com/user-attachments/files/17631153/artisan-settings-mac11-20241105.txt)

**Reply by MAKOMO** [reply]
Even with your settings file I cannot reproduce this on v3.0.2 nor on the current development version. I tried both, the macOS legacy builds and the regular macOS builds.

I also checked the code and the relevant code did not change since v2.8.4.

Only thing we still can try is if the issue persists after doing a Factory Reset (you already saved your settings for reconstruction). In any case we could check your artisan.log file. You can fetch this by a click on the plus icon in the upper left corner of the Artisan window while holding the OPTION key pressed. An email in mail.app should open with the artisan.log file attached. If you could save this to a file, rename it to end in .txt and attach it here via drag-and-drop to a comment field we can have a look if we see any traces of errors or exceptions that could shed some light on this.

THANKS!

**Reply by msevette** [reply]
I did a Factory Reset, opened the same profile we've been using, selected Tools >> Analyzer >> Auto All, and everything worked as it should. However, the projected curve line does not disappear when clear is selected (which it never did in v.2.8.4 either). Then I loaded the Artisan settings file that I sent you and the issue of the data boxes flashing and then disappearing reoccured. So, there's something in my settings that is causing this. Does this mean I need to do a factory reset and rebuild Artisan how I want it from scratch? If so, I'd rather get my hands on v.2.8.4 and try it again first.

I've attached my artisan.log file, renamed to artisan-log-msevette.txt.

Thank you for your support. I'm looking forward to hearing from you.

[artisan-log-msevette.txt](https://github.com/user-attachments/files/17644557/artisan-log-msevette.txt)

**Reply by MAKOMO** [reply]
> I did a Factory Reset, opened the same profile we've been using, selected Tools >> Analyzer >> Auto All, and everything worked as it should. 

Ok. So it is something with your specific settings.

> However, the projected curve line does not disappear when clear is selected (which it never did in v.2.8.4 either). 

This is by design. This is created in the background replacing a background profile if it was loaded.

> Then I loaded the Artisan settings file that I sent you and the issue of the data boxes flashing and then disappearing reoccured. So, there's something in my settings that is causing this. Does this mean I need to do a factory reset and rebuild Artisan how I want it from scratch? If so, I'd rather get my hands on v.2.8.4 and try it again first.

Artisan v2.8.4 is not available any longer for download and will most likely not behave better as the relevant code did not change. Better to fix this for future versions, right?

>
> I've attached my artisan.log file, renamed to artisan-log-msevette.txt.
> 
> Thank you for your support. I'm looking forward to hearing from you.
> 

I made two minimal modifications to your settings file (removed cached positions of those two analyzer panes) and attached it (you need to rename it again to end in .aset). Could you test if this health this for you?

[artisan-settings-mac11-20241105-modified.aset.txt](https://github.com/user-attachments/files/17660613/artisan-settings-mac11-20241105-modified.aset.txt)

We would be interested in your artisan.log file to see if it contains a hint on what is exactly going on (we still cannot reproduce this on any machine, OS, Artisan version) to be able to fix it. Could you press

The location of the `artisan.log` file is in the folder

```
~/Library/Application Support/artisan-scope/Artisan
```

If you are in the Finder, press SHIFT+COMMAND+G and paste the path above into the box and the Finder will open that directory. See screenshots below.

<img width="517" alt="Screenshot 2024-11-07 at 09 25 01" src="https://github.com/user-attachments/assets/5ae1f1c3-3527-4ca5-8c8a-0472ef8c487a">
<img width="507" alt="Screenshot 2024-11-07 at 09 25 23" src="https://github.com/user-attachments/assets/6fb8ed82-776e-4ba9-ba25-9078a0c47d3a">


Please copy and rename all files named `artisan.log` or `artisan.log.1` to end in .txt and attach them here on GitHub via drag-and-drop to a comment field.

THANKS!

**Reply by msevette** [reply]
I did a Factory Reset, loaded the new modified file you provided (artisan-settings-mac11-20241105-modified.aset), opened the same profile we've been working with, selected Tools >> Analyzer >> Auto All, and nothing has changed--analyzer data flashes then disappears. I have attached my most recent log file for you to look at (artisan.log.txt). I'll take time this weekend and rebuild my custom setting from scratch after doing a Factory Reset in hopes of fixing this issue. Thank you for your support.
[artisan.log.txt](https://github.com/user-attachments/files/17676976/artisan.log.txt)

**Reply by roasterdave** [reply]
We were able to reproduce this and will fix it in a future build.  There is a bit of a complex interaction among some settings that leads to this happening.  Thanks for reporting it.

For the time being, you can use your existing settings and work around the problem by simply hiding the buttons when the roaster scope is OFF.  While the scope is OFF, untick the Buttons selection in the View menu.  Don't worry, your buttons will still be displayed after clicking ON or START.

**Reply by msevette** [reply]
Good deal! I'm glad I could help. Thank you for this fantastic program and for your first-class support!
